compiler/src/typed/checkertypes.re

```diff
@@ -83,13 +84,28 @@ let constant:
       switch (Literals.conv_number_int(n)) {
       | Some(n) => Ok(Const_number(Const_number_int(n)))
       | None =>
-        Error(
-          Location.errorf(
-            ~loc,
-            "Number literal %s is outside of the integer range of the Number type.",
-            n,
-          ),
-        )
+        switch (Literals.conv_bigint(n)) {
+        | Some((bigint_negative, bigint_limbs)) =>
+          Ok(
+            Const_number(
+              Const_number_bigint({
+                bigint_negative,
+                bigint_limbs,
+                bigint_rep: n,
+              }),
+            ),
+          )
+        // Should not happen, since `None` is only returned for the empty string,
+        // and that should be disallowed by the lexer
```

Comment:
```suggestion
        // and that is disallowed by the lexer
```
Just clarifying that it does disallow it and that's not something we still need to implement.